src/cdk/observers/observe-content.ts

```diff
@@ -35,6 +32,70 @@ export class MutationObserverFactory {
   }
 }
 
+
+/** A factory that creates ContentObservers. */
+@Injectable({providedIn: 'root'})
+export class ContentObserver {
+  private _observedElements = new Map<Element, {
```

Comment:
This could be a `WeakMap`

Reply:
Whoops, I made it a `Map` so I could loop on destroy and clean up, but forgot to actually do that.